### ISSUE TITLE
localfs: get_file: support links

### ISSUE THREAD

**Issue by efiop** [iterative/dvc-objects#207]
`get_file` will always use `copyfile` right now, which will try to use `reflinks`. But there are cases where one would want to try to use other link types too in a general manner.

This is needed for datafs/dvcfs to be able to download/instantiate data from cache using symlink. See https://github.com/iterative/dvc/issues/9385

**Comment by skshetry** [comment]
I don't think `get_file` should support this. Treating symlink/hardlink as a file is a dvc concept. Localfs already provides `symlink`/`hardlink` APIs to do that.

**Comment by efiop** [comment]
@skshetry Good point! I was also not a fan of it, but it does make sense from some perspectives. For now, I think I found a better way indeed.

**Comment by efiop** [comment]
Closing for now.
